mrmustard/lab_dev/transformations/base.py

```diff
@@ -202,6 +203,35 @@ def inverse(self) -> Unitary:
             name=unitary_dual.name,
         )
 
+    @property
+    def symplectic(self):
+        r"""
+        Returns the symplectic representation
+        """
+        batch = self.representation.A.shape[0]
```

Comment:
This should simplify things
```suggestion
        batch_size = self.representation.ansatz.batch_size
        return [Au2Symplectic(self.representation.A[batch, :, :]) for batch in range(batch_size)]
```